### DISCUSSION TITLE
图床扩展

### DISCUSSION THREAD

**Discussion by sysfox** [Qexo/Qexo#244] [Ideas]
结合[Picx](https://github.com/XPoet/picx)的技术，把图片压缩后上传到github仓库，再引用cdn外链到Markdown
@am-abudu

**Comment by sysfox** [comment]
token设置思路
- 当博客源文件仓库为Github时，使用init时设置的token
- 当博客源文件在Gitlab 或 私有服务器时，则设置一个新的GitHub token

**Comment by Fgaoxing** [comment]
#121

**Comment by Fgaoxing** [comment]
@sysfox

**Comment by sysfox** [comment]
1. Picx现在不使用jsdelivr
2. Picx会采用压缩算法压缩图片，使之降至Github文件上传大小限制以下
不信的话您可以去Picx项目尝试使用一下
@Fgaoxing

**Comment by am-abudu** [comment]
我将尝试添加能够上传到Github的服务商，但考虑Vercel的性能问题，不会对图片压缩

**Reply by sysfox** [reply]
不能把这个功能集成到Qexo中吗？
@am-abudu

**Reply by am-abudu** [reply]
当然是集成的，对于图片的压缩以后可能做到前端

**Reply by sysfox** [reply]
Picx也可以部署在vercel，但它似乎是在前端压缩图片的

**Reply by am-abudu** [reply]
目前添加 托管平台/图床/推送器 都是以服务商模块的方式模块化接入

**Reply by sysfox** [reply]
你可以试试研究下Picx的源码

**Reply by am-abudu** [reply]
前端压缩图片要考虑一下逻辑，并且我现在时间不够

**Reply by sysfox** [reply]
哈哈，加油！

**Comment by sysfox** [comment]
已发现picx图片压缩核心代码
https://github.com/XPoet/picx/blob/master/src/utils/compress.ts
@am-abudu

**Reply by sysfox** [reply]
依赖仓库：
https://github.com/myeveryheart/squoosh-browser

**Comment by Fgaoxing** [comment]
@am-abudu GitHub允许吗？

**Reply by sysfox** [reply]
我个人认为可以，在您提供的链接中，Github只是限制了文件上传大小
Github条款中没有提出禁止把仓库作为图床
https://docs.github.com/cn/site-policy/

**Comment by am-abudu** [comment]
已经在Dev分支给予支持

**Comment by sysfox** [comment]
前端压缩图片的代码可以提上日程了吗？doge
[#discussioncomment-4214891](#discussioncomment-4214891)
@am-abudu

**Reply by am-abudu** [reply]
没有想到很周全的方案，你发的项目在这里也显得不是很好用，如果压缩成Webp很多地方无法上传

**Reply by sysfox** [reply]
> 压缩成Webp很多地方无法上传

能具体解释下是什么意思吗？是否是指无法上传到图床?(应该不会吧)

**Reply by am-abudu** [reply]
是说有一些图床并不支持webp格式的上传。另外主要问题是我没找到合适的前端库

**Reply by sysfox** [reply]
有图床不支持就离谱

**Comment by sysfox** [comment]
那我就再等等吧

**Reply by am-abudu** [reply]
我暂时不会支持这一项功能，如果你使用的OSS/CDN有图片相关功能可以通过链接后缀的形式完成这项操作
